aten/src/ATen/core/dispatch/DispatchKeyExtractor.h

```diff
@@ -46,6 +47,15 @@ static inline DispatchKeySet computeDispatchKeySet(
   return (((ks | local.included_) - local.excluded_) & key_mask);
 }
 
+// NB: This is also used in backend select
+inline void updateDispatchKeySetFromSize(DispatchKeySet& ks, c10::SymIntArrayRef sizes) {
+  for (const auto& x : sizes) {
+    if (x.is_heap_allocated()) {
+      ks = ks | x.toSymNode()->key_set();
```

Comment:
In fact, it looks like you can potentially just read this off of `key_set`

Reply:
Updated, there was a little more logic that needed to be added in python_symnode to make this happen, but overall we're should be able to avoid a lot of roundtrips to python.